spec/migrations/20151021093644_set_correct_sti_type_on_cloud_network_spec.rb

```diff
@@ -0,0 +1,39 @@
+require_migration
+
+describe SetCorrectStiTypeOnCloudNetwork do
+  let(:cloud_manager_stub)   { migration_stub(:CloudManager) }
+  let(:cloud_network_stub)   { migration_stub(:CloudNetwork) }
+
+  let!(:empty_cloud_network) { cloud_network_stub.create! }
+
+  let!(:ems_cloud)           { cloud_manager_stub.create!(:type => "ManageIQ::Providers::Openstack::CloudManager") }
+  let!(:cloud_cloud_network) { cloud_network_stub.create!(:external_facing => true, :ems_id => ems_cloud.id) }
+
+  let!(:ems_infra)           { cloud_manager_stub.create!(:type => "ManageIQ::Providers::Openstack::InfraManager") }
+  let!(:infra_cloud_network) { cloud_network_stub.create!(:external_facing => false, :ems_id => ems_infra.id) }
+
+  migration_context :up do
+    it "sets correct type to cloud_network objects according to cloud_manager" do
+      migrate
+
+      expect(empty_cloud_network.reload.type).to eq(described_class::CLOUD_TEMPLATE_CLASS)
```

Comment:
Are you trying to verify that a record created before the migration gets the proper type after the migration?  If so, then this test isn't doing that.  The empty_cloud_network record is not created until this line, but this is after the migration has occurred.